resources/js/electron-builder.js

```diff
@@ -12,28 +12,48 @@ const appUrl = process.env.APP_URL;
 const appAuthor = process.env.NATIVEPHP_APP_AUTHOR;
 const phpBinaryPath = process.env.NATIVEPHP_PHP_BINARY_PATH;
 const certificatePath = process.env.NATIVEPHP_CERTIFICATE_FILE_PATH;
-const isArm64 = process.argv.includes('--arm64');
-const isWindows = process.argv.includes('--win');
-const isLinux = process.argv.includes('--linux') || 'linux' === os.platform();
-const isDarwin = process.argv.includes('--mac');
-let targetOs = 'mac';
+
 let phpBinaryFilename = 'php';
-if (isWindows) {
-    targetOs = 'win';
-    phpBinaryFilename += '.exe';
-}
-if (isLinux) {
-    targetOs = 'linux';
-}
 
-let binaryArch = 'x86';
-if (isArm64) {
-    binaryArch = 'arm64';
-}
-if (isWindows || isLinux) {
-    binaryArch = 'x64';
+// Allows us to map the platform name to the directory name for development mode
+const platformDirectory = {
+    win32: 'win',
+    darwin: 'mac',
+    linux: 'linux',
+};
+
+// These are the available platforms we can build for
+const platforms = ['win', 'mac', 'linux'];
+
+// Default to the current platform for develop mode. Build will pass in an arg that overrides this
+let targetOs = platformDirectory[process.platform];
+
+// Default to the current arch for develop mode.
+// Build will default to x64 but can be set to arm64
+let binaryArch = process.arch;
+
+// If we're building, we need to check for target overrides
+if (isBuilding) {
+    // Check for a target platform flag
+    for (const platform of platforms) {
+        if (process.argv.includes('--' + platform)) {
+            targetOs = platform;
+            break;
+        }
+    }
+
+	// Check for forced ARM build, else default to x64
+    if (process.argv.includes('--arm64')) {
+        binaryArch = 'arm64';
+    } else {
+        binaryArch = 'x64';
+    }
 }
 
+// Add .exe to the filename if we're on Windows
```

Comment:
We do this check after so that whether we're building for development on Windows OR targeting the build to Windows, it will set the binary extension.